src/main/java/com/coyoapp/tinytask/repository/TaskRepository.java

```diff
@@ -2,6 +2,14 @@
 
 import com.coyoapp.tinytask.domain.Task;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import java.util.List;
 
 public interface TaskRepository extends JpaRepository<Task, String> {
+
+  @Query(value = "SELECT * FROM task t ORDER BY duedate ASC",
```

Comment:
Also just a hint as this is backend code: A custom query is not needed here, you could have called the existing JpaRepository method findAll(Sort...) with a sort parameter.

Reply:
So basically I tried JpaRepository method to sort in a way that all duadate with null values comes after all non null values, which is not working in a way I'm expecting, so that's why I used the custom query.